server/publications/collections/product-publications.app-test.js

```diff
@@ -152,166 +206,195 @@ describe("Publication", function () {
         sandbox.stub(Reaction, "hasPermission", () => true);
         sandbox.stub(Reaction, "getShopsWithRoles", () => [shopId]);
 
-        let isDone = false;
-
         collector.collect("Products", 24, undefined, {}, (collections) => {
           const products = collections.Products;
           const data = products[1];
           const expectedTitles = ["My Little Pony", "Shopkins - Peachy"];
 
           expect(expectedTitles.some((title) => title === data.title)).to.be.ok;
-
-          if (!isDone) {
-            isDone = true;
-            done();
-          }
-        });
+        }).then(() => done(/* empty */), done);
       });
 
       it("should return only visible products to visitors", function (done) {
         sandbox.stub(Reaction, "getShopId", () => shopId);
         sandbox.stub(Roles, "userIsInRole", () => false);
 
-        let isDone = false;
-
         collector.collect("Products", 24, undefined, {}, (collections) => {
           const products = collections.Products;
           const data = products[0];
           const expectedTitles = ["Fresh Tomatoes", "Shopkins - Peachy"];
 
-          // the correct expectation should be 2, but there is an issue where
```

Comment:
ignore this comment.
this PR is not against master, but a branch I created to ensure that admins see only the shops/products for which they have edit rights.